### DISCUSSION TITLE
Storage usage of mon Pod's PV is increasing

### DISCUSSION THREAD

**Discussion by ytmt365** [rook/rook#14428] [Q&A]
Hi,

I have deployed cephcluster with 3 monitoring Pods like below, and storage usage of those PVs are increasing.

```console
$ oc get pod,pvc -l app=rook-ceph-mon
NAME                                   READY   STATUS    RESTARTS      AGE
pod/rook-ceph-mon-a-556bcb7dcf-cskfg   1/1     Running   0             20d
pod/rook-ceph-mon-b-7796764c4d-n7s25   1/1     Running   1 (20d ago)   20d
pod/rook-ceph-mon-d-6c77dc5db8-6hf6n   1/1     Running   0             20d

NAME                                    STATUS   VOLUME              CAPACITY   ACCESS MODES   STORAGECLASS    AGE
persistentvolumeclaim/rook-ceph-mon-a   Bound    csivol-39fd063a60   10Gi       RWO            local-storage   20d
persistentvolumeclaim/rook-ceph-mon-b   Bound    csivol-0608104871   10Gi       RWO            local-storage   20d
persistentvolumeclaim/rook-ceph-mon-d   Bound    csivol-4f7d597843   10Gi       RWO            local-storage   20d
```

OpenShift: `4.12.44`
Rook: `v1.12.2`
Ceph: `v17.2.6`

Is it possible that storage usage of mon Pod's PV fills up?
If so, how large should PV size be set?

**Comment by parth-gr** [comment]
How much is the used capacity?

**Reply by ytmt365** [reply]
Management software of my storage appliance shows 82% (8.2 Gi) in use.

**Reply by ytmt365** [reply]
@parth-gr 
However, running `du` command in my pod shows only 57MB in use.

```console
$ oc exec -it rook-ceph-mon-a-556bcb7dcf-cskfg -c mon -- du -h /var/lib/ceph/mon/ceph-a
57M	/var/lib/ceph/mon/ceph-a/store.db
57M	/var/lib/ceph/mon/ceph-a
```

**Reply by parth-gr** [reply]
can you check the logrotate-container files inside the mon pod, I think they might be increasing?

**Reply by ytmt365** [reply]
@parth-gr 
Could you tell me how to check that?  
Can I check that from result of `oc get pod rook-ceph-mon-a-556bcb7dcf-cskfg -oyaml` ?

**Reply by parth-gr** [reply]
oc exec -it rook-ceph-mon-a-556bcb7dcf-cskfg -c  log-collector

Now check the file sizes

**Reply by ytmt365** [reply]
Thank you for your instruction. But there's no logrotate-container in the mon Pod.

```console
$ oc exec -it rook-ceph-mon-a-556bcb7dcf-cskfg -c log-collector -- du -h /
Error from server (BadRequest): container log-collector is not valid for pod rook-ceph-mon-a-556bcb7dcf-cskfg
```

**Comment by travisn** [comment]
You may want to increase the mon PVC size to 50Gi, which would be plenty large. Or you could adjust the log rotation sizes as Parth is have you check on that.

**Reply by ytmt365** [reply]
Thank you for your advice.
Is 50Gi large enough? Is it not possible that the storage usage reaches up to near 50Gi in regular cases?

**Reply by travisn** [reply]
The mons only contain metadata that is not expected to grow much. I have not heard of any issues in production with 50Gi for the mons.
